### PR TITLE
Task/run validation pre training

### PR DESCRIPTION
**Pull request by badisa**

### CODE REVIEW

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 1]
```diff
@@ -311,8 +339,13 @@ def unflatten(theta: array) -> Dict[Handler, array]:
     np.save(output_path.joinpath('step_indices.npy'), np.hstack(step_inds)[:args.param_updates])
 
     step = 0
+    epoch = 0
     # in each optimizer step, look at one transformation from relative_transformations
-    for epoch, steps in enumerate(step_inds):
+    for steps in step_inds:
+        # Run Validation edges at start of epoch. Unlike NNs we have a reasonable starting
+        # point that is worth knowing
+        run_validation_edges(validation, ordered_params, systems, epoch)
```
Comment (badisa) [inline examples/hif2a/fit_to_multiple_rbfes.py:347]: Running the validation after training made it hard to determine the training without rerunning RBFE calculations.

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 2]
```diff
@@ -333,15 +366,17 @@ def unflatten(theta: array) -> Dict[Handler, array]:
...
             print(f"at optimizer step {step}, loss={loss:.3f}")
 
             # check if it's probably okay to take an optimizer step on the basis of this result
             # TODO: move responsibility for returning error flags / simulation uncertainty estimates further upstream
-            blown_up = any(_blew_up(results) for results in results_this_step.values())
+            blown_up = False
+            for stage, results in results_this_step.items():
+                if _blew_up(results):
+                    blown_up = True
+                    print(f"step {step} blew up in {stage} stage")
```
Comment (badisa) [inline examples/hif2a/fit_to_multiple_rbfes.py:379]: Here just to make it clear where exactly the simulation blew up.

File: examples/hif2a/fit_to_multiple_rbfes.py (hunk 1 above)
Comment (proteneer) [reply examples/hif2a/fit_to_multiple_rbfes.py:347]: +1, running validation before hand is the standard way to do this.

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 3]
```diff
@@ -164,6 +164,36 @@ def unflatten(theta: array) -> Dict[Handler, array]:
     return theta, unflatten
 
 
+def run_validation_edges(validation: Dataset, params, systems, epoch):
+    if len(validation) <= 0:
+        return
+    val_loss = np.zeros(len(validation))
+    for i, rfe in enumerate(validation.data):
+        if getattr(rfe, "complex_path", None) is not None:
```
Comment (proteneer) [inline examples/hif2a/fit_to_multiple_rbfes.py:172]: This looks a little janky

File: fe/model.py [hunk 4]
```diff
@@ -93,7 +93,7 @@ def predict(self, ff_params: list, mol_a: Chem.Mol, mol_b: Chem.Mol, core: np.nd
 
             x0 = coords
             v0 = np.zeros_like(coords)
-            box = np.eye(3, dtype=np.float64)*100 # note: box unused
+            # box = np.eye(3, dtype=np.float64)*100 # note: box unused
```
Comment (proteneer) [inline fe/model.py:96]: We can probably just remove this line as opposed to commenting it out.

**Review by proteneer — APPROVED** [review]

File: examples/hif2a/fit_to_multiple_rbfes.py (hunk 3 above)
Comment (badisa) [reply examples/hif2a/fit_to_multiple_rbfes.py:172]: It is, and can probably go away. Had it as I had older transformations that were generated that didn't have an associated path so I would default to the one provided by the script. The same logic happens for the training code

**Review by maxentile — APPROVED** [review]
Nice! This will allow us to address questions about the difference in performance before and after training without having to choose between the hassle of benchmarking the initial force field in a separate pass, or making the unsafe assumption that "performance before training" = "performance after 1 epoch"

**Pull request merged**